client_test/image_test.py

```diff
@@ -302,16 +303,23 @@ def test_image_build_with_context_mount(client, servicer, tmp_path):
     data_mount = Mount(local_dir=tmp_path, remote_dir="/")
 
     stub = Stub()
-    stub["image"] = Image.debian_slim().copy(data_mount, remote_path="/dummy")
+    dockerfile = NamedTemporaryFile("w", delete=False)
+    dockerfile.write("COPY . /dummy\n")
+    dockerfile.close()
+
+    stub["copy"] = Image.debian_slim().copy(data_mount, remote_path="/dummy")
+    stub["from_dockerfile"] = Image.debian_slim().dockerfile_commands(["COPY . /dummy"], context_mount=data_mount)
+    stub["dockerfile_commands"] = Image.debian_slim().from_dockerfile(dockerfile.name, context_mount=data_mount)
 
     with stub.run(client=client) as running_app:
-        layers = get_image_layers(running_app["image"].object_id, servicer)
-        assert "COPY . /dummy" in layers[0].dockerfile_commands
-        assert layers[0].context_mount_id == "mo-123"
-        files = {f.rel_filename: f.content for f in data_mount._get_files()}
-        assert files["data.txt"] == b"hello"
-        assert files[os.path.join("data", "sub")] == b"world"
-        assert len(files) == 2
+        for (image_name, expected_layer) in [("copy", 0), ("dockerfile_commands", 1), ("from_dockerfile", 0)]:
+            layers = get_image_layers(running_app[image_name].object_id, servicer)
+            assert layers[expected_layer].context_mount_id == "mo-123", f"error in {image_name}"
+            assert "COPY . /dummy" in layers[expected_layer].dockerfile_commands
+            files = {f.rel_filename: f.content for f in data_mount._get_files()}
```

Comment:
This loop looks to be re-testing the same `data_mount` object which is not related to the 3 image objects. From poking into the code, I think it's `Image.context_files` that has the data to assert on?

Reply:
Oh yeah the data_mount part shouldn't be part of the loop (since the same is used for all three images). good catch.
Image.context_files is something else though, that's the legacy payload for attaching context files (now mostly used for the modal requirements.txt)